### PR TITLE
Adding UT SpatialStructure-4

### PR DESCRIPTION
**Pull request by larswik**
## Unit test

This UT establishes a spatial structure for the level crossing storyline. The intention is to serve as a skeleton for upcoming unit tests for this storyline.

It is based on the storylines [SLLC-DD](https://app.box.com/folder/119147119179?s=q1x0vz7yqq7otrlet7dm2dx4u44s8fks) (IFC Rail phase 2) and [bSFi_IR_SLRD_DD](https://app.box.com/folder/122373280942?s=x9q3q62tcc77hdqpdw0vjncj7bg3byay) (IFC Infra Deployment)

## Covered topics

- [x] project setup
- [x] georeferencing
- [ ] alignment semantics
- [ ] alignment geometry
- [ ] linear placement
- [x] project breakdown structure / spatial structure
- [ ] terrain
- [ ] triangulated irregular network
- [ ] open cross profile
- [ ] sectioned surface & solid
- [ ] pavement & course
- [ ] surface feature
- [ ] earthworks
- [ ] geotechnics
- [ ] longitudinal products (barriers, guardrail, ...)
- [ ] interchange
- [ ] structural
- [x] drainage
- [ ] signage
- [ ] road furniture
- [ ] rail furniture
- [ ] ports & waterways furniture
- [ ] bridge furniture

### CODE REVIEW

File: SpatialStructure-4/readme.md [hunk 1]
```diff
@@ -28,13 +28,13 @@ This scenario covers the following concepts and/or IFC entities:
 
 The figure below shows the extent of the IfcSite and its georeferenced location.
 
-![](./Figure 1 - Situation.PNG)
+![](./Figure%201%20-%20Situation.PNG)
```
Comment (pjanck) [inline SpatialStructure-4/readme.md]: Suggestion: What about not using spaces in filenames?

**Review by pjanck — CHANGES_REQUESTED** [review]

File: SpatialStructure-4/readme.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+### Prerequisites
+
+None
+
+### Content
+
+This scenario covers the following concepts and/or IFC entities:
+- Georeferencing
+  - `IfcProjectedCRS`
+  - `IfcMapConversion`
```
Comment (pjanck) [inline SpatialStructure-4/readme.md]: I'd rather see this as a dependency to Georeferencing-1.

File: SpatialStructure-4/readme.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+          - InterferenceType=Crosses
+          - RelatingElement=#A.a.2.1
+          - RelatedElement=#A.a.4.4
+
+_* the PredefinedType enumeration is selected through "IfcFacilityPartTypeSelect"_.
+
+
+---
+
+#### Expected Results
```
Comment (pjanck) [inline SpatialStructure-4/readme.md]:
```suggestion
### Expected Results
```

File: SpatialStructure-4/readme.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        - UsageType=LONGITUDINAL
+        - IfcRelInterferesElements
+          - InterferenceType=Crosses
+          - RelatingElement=#A.a.2.1
+          - RelatedElement=#A.a.4.4
+
+_* the PredefinedType enumeration is selected through "IfcFacilityPartTypeSelect"_.
+
+
+---
```
Comment (pjanck) [inline SpatialStructure-4/readme.md]: This will be fun seeing in the resulting PDF.

I'd suggest removing it altogether.

**Review by pjanck — CHANGES_REQUESTED** [review]
Readme suggestions.

File: SpatialStructure-4/readme.md (hunk 2 above)
Comment (larswik) [reply SpatialStructure-4/readme.md]: I am not sure that I understand? A dependency in what way? It uses Georeferencing concepts, but no direct link to that unit test??

File: SpatialStructure-4/readme.md (hunk 4 above)
Comment (larswik) [reply SpatialStructure-4/readme.md]: Ok. You mean the loooong list?

File: SpatialStructure-4/readme.md (hunk 1 above)
Comment (larswik) [reply SpatialStructure-4/readme.md]: Sure - will correct that.

Comment (larswik) [reply SpatialStructure-4/readme.md]: Fixed

File: SpatialStructure-4/readme.md (hunk 3 above)
Comment (larswik) [reply SpatialStructure-4/readme.md]: Fixed

File: SpatialStructure-4/readme.md (hunk 4 above)
Comment (larswik) [reply SpatialStructure-4/readme.md]: Removed the list altogether.

File: SpatialStructure-4/readme.md (hunk 2 above)
Comment (pjanck) [reply SpatialStructure-4/readme.md]: A link to a unit test does not mean, that the same *content* is being built upon, but that the same *concepts* are being used. It means, that in order to implement this unit test, it is advisable to first implement the other one.

File: SpatialStructure-4/readme.md (hunk 4 above)
Comment (pjanck) [reply SpatialStructure-4/readme.md]: No, I mean the horizontal line :D

File: SpatialStructure-4/readme.md (hunk 2 above)
Comment (larswik) [reply SpatialStructure-4/readme.md]: But should it then reference all georeferencing UT:s (if there are more than one)? I would then consider it a reference to the topic and not a specific unit test?

File: SpatialStructure-4/readme.md (hunk 4 above)
Comment (larswik) [reply SpatialStructure-4/readme.md]: Oops...

File: SpatialStructure-4/readme.md (hunk 2 above)
Comment (pjanck) [reply SpatialStructure-4/readme.md]: I see your point. It is difficult to visualize it in the graph, though. Additionally: the other goereferencing UT bring in different concepts that you probably don't need in this UT. So I'd go with the "minimal" requirement needed.

File: SpatialStructure-4/readme.md (hunk 4 above)
Comment (pjanck) [reply SpatialStructure-4/readme.md]: I quite liked the list, please reverse the content! :)

Comment (larswik) [reply SpatialStructure-4/readme.md]: Just to be really clear - which horizontal line do you mean?

Comment (pjanck) [reply SpatialStructure-4/readme.md]: I mean the three `---` there.

Comment (larswik) [reply SpatialStructure-4/readme.md]: I hope that they are gone...

Comment (pjanck) [reply SpatialStructure-4/readme.md]: Yes, looks good now.

File: SpatialStructure-4/readme.md (hunk 2 above)
Comment (pjanck) [reply SpatialStructure-4/readme.md]: Perhaps the Project Setup 1 would be a correct "parent"?

**Pull request merged**